libbeat/cmd/instance/beat.go

```diff
@@ -741,8 +752,71 @@ func (b *Beat) registerTemplateLoading() error {
 			return err
 		}
 
-		if esCfg.Index != "" && (cfg.Name == "" || cfg.Pattern == "") && (b.Config.Template == nil || b.Config.Template.Enabled()) {
-			return fmt.Errorf("setup.template.name and setup.template.pattern have to be set if index name is modified.")
+		if esCfg.Index != "" && (templateCfg.Name == "" || templateCfg.Pattern == "") && (b.Config.Template == nil || b.Config.Template.Enabled()) {
+			return fmt.Errorf("setup.template.name and setup.template.pattern have to be set if index name is modified")
+		}
+
+		if b.Config.ILM.Enabled() {
+
+			cfgwarn.Beta("Index lifecycle management is enabled which is in beta.")
+			logp.Info("Overwriting setup.template for ILM")
+
+			ilmCfg, err := getILMConfig(b)
+			if err != nil {
+				return err
+			}
+
+			// Template name and pattern can't be configure when using ILM
+			logp.Info("Overwriting template name and pattern as ILM is enabled.")
+
+			// In case no template settings are set, config must be created
+			if b.Config.Template == nil {
+				b.Config.Template = common.NewConfig()
+			}
+			b.Config.Template.SetString("name", -1, ilmCfg.RolloverAlias)
+			b.Config.Template.SetString("pattern", -1, fmt.Sprintf("%s-*", ilmCfg.RolloverAlias))
+
+			// rollover_alias and lifecycle.name can't be configured and will be overwritten
+			logp.Info("Overwrite template settings for rollover_alias and lifecycle.name.")
+			err = b.Config.Template.SetString("settings.index.lifecycle.rollover_alias", -1, ilmCfg.RolloverAlias)
+			if err != nil {
+				logp.Err("error setting rollover_alias setting: %s", err)
```

Comment:
nit: Log message should start with upper case

Reply:
fixed